src/gui/torrentoptionsdialog.cpp

```diff
@@ -343,8 +454,79 @@ int TorrentOptionsDialog::getSeedingTime() const
     return m_ui->spinTimeLimit->value();
 }
 
+void TorrentOptionsDialog::handleCategoryChanged(const int index)
+{
+    Q_UNUSED(index);
+
+    if (m_ui->checkAutoTMM->checkState() == Qt::Checked)
+    {
+        if (!m_allSameCategory && (m_ui->comboCategory->currentIndex() == 0))
+        {
+            m_ui->savePath->setSelectedPath(QLatin1String(""));
+        }
+        else
+        {
+            const QString savePath = BitTorrent::Session::instance()->categorySavePath(m_ui->comboCategory->currentText());
+            m_ui->savePath->setSelectedPath(Utils::Fs::toNativePath(savePath));
+        }
+    }
+    else if (m_ui->checkAutoTMM->checkState() == Qt::PartiallyChecked)
+    {
+        m_ui->savePath->setSelectedPath(QLatin1String(""));
+    }
+
+    if (!m_allSameCategory && (m_ui->comboCategory->currentIndex() == 0))
+    {
+        m_ui->comboCategory->clearEditText();
+        m_ui->comboCategory->lineEdit()->setPlaceholderText(m_currentCategoriesString);
+    }
+    else
+    {
+        m_ui->comboCategory->lineEdit()->setPlaceholderText(QLatin1String(""));
+    }
+}
+
+void TorrentOptionsDialog::handleTMMChanged()
+{
+    if (!m_ui->checkAutoTMM->isChecked())
```

Comment:
Anyway you should use `checkState()` here to avoid confusion.

Reply:
>Does isChecked() return true in case check box is partially checked?

Yes, i only found out about it in this PR, which is why i started using `checkState()`. Everything i've done in this dialog was under the assumption that `isChecked()` is true only for checked. Fortunately it doesn't seem to matter in other cases (by chance).

>Anyway you should use checkState() here to avoid confusion.

Done.